<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScriptControl</class>
 <widget class="QWidget" name="ScriptControl">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>725</width>
    <height>415</height>
   </rect>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QSplitter" name="verticalSplitter">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <widget class="QSplitter" name="horizontalSplitter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>5</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <widget class="JSCodeEditor" name="codeEditor">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>2</horstretch>
         <verstretch>4</verstretch>
        </sizepolicy>
       </property>
       <property name="plainText">
        <string>Script.onInit(()=&gt;{
	Server.addressBase = AddressBase.Base1;
	Server.onChange(1/* deviceId */, Register.Holding, 1, (value)=&gt;
	{
		/* write your code on register value change */
	});

	/* write your init code here */
});</string>
       </property>
       <property name="tabStopDistance">
        <double>20.000000000000000</double>
       </property>
       <property name="placeholderText">
        <string/>
       </property>
      </widget>
      <widget class="ConsoleOutput" name="console">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>1</verstretch>
        </sizepolicy>
       </property>
       <property name="undoRedoEnabled">
        <bool>false</bool>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
     <widget class="HelpWidget" name="helpWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>JSCodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>jscodeeditor.h</header>
  </customwidget>
  <customwidget>
   <class>ConsoleOutput</class>
   <extends>QPlainTextEdit</extends>
   <header>consoleoutput.h</header>
  </customwidget>
  <customwidget>
   <class>HelpWidget</class>
   <extends>QTextBrowser</extends>
   <header>helpwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
